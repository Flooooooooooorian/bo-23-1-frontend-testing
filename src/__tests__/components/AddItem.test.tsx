import {fireEvent, render, screen} from "@testing-library/react";
import AddItem from "../../components/AddItem";

describe("AddItem Tests", () => {

    test("renders correctly", () => {
        //GIVEN
        const addItemFunction = () => {}

        //WHEN
        const view = render(<AddItem addItem={addItemFunction} />)

        //THEN
        expect(view.asFragment()).toMatchSnapshot()
    })

    test("add item", () => {
        //GIVEN
        const mockAddItem = jest.fn()

        //WHEN
        const view = render(<AddItem addItem={mockAddItem} />)

        //THEN
        expect(view.asFragment()).toMatchSnapshot()
        
        //WHEN
        fireEvent.input(screen.getByPlaceholderText("Name"), {target: {value: "test-input"}})

        //THEN
        expect(view.asFragment()).toMatchSnapshot()

        //WHEN
        fireEvent.click(screen.getByText("Save"))

        //THEN
        expect(view.asFragment()).toMatchSnapshot()

        expect(mockAddItem).toBeCalledWith("test-input")


    })
})
