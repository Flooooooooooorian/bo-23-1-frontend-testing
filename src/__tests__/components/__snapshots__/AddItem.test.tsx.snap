// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddItem Tests add item 1`] = `
<DocumentFragment>
  <input
    placeholder="Name"
    value=""
  />
  <button>
    Save
  </button>
</DocumentFragment>
`;

exports[`AddItem Tests add item 2`] = `
<DocumentFragment>
  <input
    placeholder="Name"
    value="test-input"
  />
  <button>
    Save
  </button>
</DocumentFragment>
`;

exports[`AddItem Tests add item 3`] = `
<DocumentFragment>
  <input
    placeholder="Name"
    value="test-input"
  />
  <button>
    Save
  </button>
</DocumentFragment>
`;

exports[`AddItem Tests renders correctly 1`] = `
<DocumentFragment>
  <input
    placeholder="Name"
    value=""
  />
  <button>
    Save
  </button>
</DocumentFragment>
`;
